desktop-widgets/configuredivecomputerdialog.cpp

```diff
@@ -14,21 +14,41 @@
 #include <QProgressDialog>
 #include <QSettings>
 
+static int getTypeFromVendorProduct(QString vendor, QString product)
+{
+	if (vendor == "Heinrichs Weikamp") {
+		if (product == "OSTC Mk2" ||
+		    product == "OSTC 2N" ||
+		    product == "OSTC 2C") {
+			return 0;
+		}
+		if (product == "OSTC 2" ||
+		    product == "OSTC 3" ||
+		    product == "OSTC Sport" ||
+		    product == "OSTC Plus" ||
+		    product == "OSTC cR") {
+			return 1;
+		}
+		if (product == "OSTC 4") {
+			return 2;
+		}
+		return -1;
+	}
+	if (vendor == "Suunto" &&
+	    (product == "Vyper" ||
+	     product == "Stinger" ||
+	     product == "Mosquito" ||
+	     product == "D3" ||
+	     product == "Vytec" ||
+	     product == "Cobra" ||
+	     product == "Gekko" ||
+	     product == "Zoop"))
+		return 3;
+	return -1;
+}
```

Comment:
Rather than hard coding a bunch of names, whitelist DC_FAMILY_HW_OSTC, DC_FAMILY_HW_OSTC3, and DC_FAMILY_SUUNTO_VYPER.

For odd reasons, the OSTC 4 is in the DC_FAMILY_HW_OSTC3 family, but for this perspective, it works differently.

Reply:
At the point when I make this determination, I have the two strings.
But yeah, I guess I should parse that into a family and use that instead.
Or just not bother. I don't use this. Ever. Why am I fixing it?